.github/workflows/gh-pages.yml

```diff
@@ -1,30 +1,55 @@
 # This workflow aims to automatically rebuild eamxx documentation
-# every time the master branch is updated on github
-name: Build and deploy gh-pages branch with Mkdocs
+# every time the master branch is updated on github and within every PR
+
+name: EAMxx Docs
 
 on:
   # Runs every time master branch is updated
   push:
-    branches: ["master"]
+    branches: [ master ]
+  # Runs every time master a PR is open against master
+  pull_request:
+    branches: [ master ]
```

Comment:
The idea of being able to display the docs before we merge is nice. I am not sure where they could be displayed, since there is only one gh page offered by github, but I'm sure there is some action that serves the doc in a temporary server. Maybe peaceiris/actions-gh-pages` can do that, somehow.

Reply:
Turns out readthedocs can do this: https://docs.readthedocs.io/en/stable/pull-requests.html

Reply:
that would replace mkdocs, right? I am by no means attached to mkdocs per se, but want to first ensure all current features we need/want will be supported if we switch.

Reply:
No it wouldn't replace mkdocs.  mkdocs would still generate the pages but they would be published to readthedocs (in addition to github.io)